sdks/python/apache_beam/io/tfrecordio.py

```diff
@@ -178,6 +180,44 @@ def read_records(self, file_name, offset_range_tracker):
           yield self._coder.decode(record)
 
 
+def _create_tfrcordio_source(
+    file_pattern=None, coder=None, compression_type=None,
+    validate=False):
+  return _TFRecordSource(file_pattern, coder, compression_type, validate)
+
+
+class ReadAllFromTFRecord(PTransform):
+  """A ``PTransform`` for reading a ``PCollection`` of TFRecord files."""
+
+  def __init__(
+      self,
+      coder=coders.BytesCoder(),
+      compression_type=CompressionTypes.AUTO,
+      **kwargs):
+    """Initialize the ``ReadAllFromText`` transform.
+
+    Args:
+      coder: Coder used to decode each record.
+      compression_type: Used to handle compressed input files. Default value
+          is CompressionTypes.AUTO, in which case the file_path's extension will
+          be used to detect the compression.
+      **kwargs: optional args dictionary. These are passed through to parent
+        constructor.
+    """
+    super(ReadAllFromTFRecord, self).__init__(**kwargs)
+    source_from_file = partial(
+        _create_tfrcordio_source, compression_type=compression_type,
+        coder=coder)
+    self._compression_type = compression_type
+    # Desired and min bundle sizes do not matter since TFRecord files are
+    # unsplittable.
+    self._read_all_files = ReadAllFiles(
+        False, compression_type, 0, 0, source_from_file)
```

Comment:
<!--new_thread; commit:d4254d9fe817a78ccd0f7704c87bc89f903add8c; resolved:0-->
Could you add keyword names here? It's hard to tell what the 0's and False are.

Reply:
Done.